### PR TITLE
Mobileapps 1311

### PR DESCRIPTION
**Pull request by aman-alfresco**
- optimized code for components
- adds focus on chips for talkback
- adds task view screen
- adds a recent comment on view screen
- adds up to 4 attachments on the view screen
- adds analytics for the task view screen
- adds accessibility for the task view screen

### CODE REVIEW

File: component/src/main/java/com/alfresco/content/component/StringExt.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    return if (multipleValue)
+        context.getString(R.string.name_truncate_in, this.take(5), this.takeLast(5))
+    else
+        context.getString(R.string.name_truncate_end, this.take(chipTextDisplayLimit))
+}
+
+/**
+ * returns formatted date string for query
+ */
+fun String.getQueryFormat(): String {
```
Comment (gl-lovekesh) [inline component/src/main/java/com/alfresco/content/component/StringExt.kt:38]: Date formatter should move to date extension.

File: listview/src/main/kotlin/com/alfresco/content/listview/tasks/TaskPriority.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * Marked as TaskPriority enum class
+ */
+enum class TaskPriority(val value: String) {
+    LOW("low"),
+    MEDIUM("medium"),
+    HIGH("high")
+}
+
+private fun getTaskPriority(priority: String): TaskPriority {
```
Comment (gl-lovekesh) [inline listview/src/main/kotlin/com/alfresco/content/listview/tasks/TaskPriority.kt:16]: match the cases as required and invalid cases move in else.

**Review by gl-lovekesh — CHANGES_REQUESTED** [review]

File: component/src/main/java/com/alfresco/content/component/StringExt.kt (hunk 1 above)
Comment (aman-alfresco) [reply component/src/main/java/com/alfresco/content/component/StringExt.kt]: Done

File: listview/src/main/kotlin/com/alfresco/content/listview/tasks/TaskPriority.kt (hunk 2 above)
Comment (aman-alfresco) [reply listview/src/main/kotlin/com/alfresco/content/listview/tasks/TaskPriority.kt]: Done

**Review by gl-lovekesh — APPROVED** [review]

**Pull request merged**